src/main/java/duke/command/SaveCommand.java

```diff
@@ -0,0 +1,42 @@
+package duke.command;
+
+import java.io.IOException;
+
+import duke.DukeException;
+import duke.task.TaskList;
+import duke.util.Storage;
+import duke.util.Ui;
+
+/**
+ * A command to save all current tasks to the disk.
+ * @author Junyi
+ */
+public class SaveCommand extends Command {
+
+    private final TaskList taskList;
+    private final Storage storage;
+    private final Ui ui;
+
+    /**
+     * Constructor for SaveCommand.
+     * Saves the tasks to local storage.
+     * @param taskList TaskList of Duke's tasks.
+     * @param storage Storage instance of Duke.
+     * @param ui Ui instance of Duke.
+     */
+    public SaveCommand(TaskList taskList, Storage storage, Ui ui) {
+        this.taskList = taskList;
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    @Override
+    public String execute() throws DukeException {
+        try {
+            storage.save(taskList);
+            return ui.showSavedDataMessage();
+        } catch (IOException e) {
+            return "Arii can't access your files... Fix your system first.";
+        }
+    }
+}
```

Comment:
If I'm not mistaken, it seems like you have implemented Arli in such a way where the user has to enter save before any of the changes made will be saved. For me, I just included this save method at the end of every method in TaskList that made a change to the TaskList, which was why I didn't have a SaveCommand class and if my Duke was to terminate by accident for some reason halfway through, the data inputted would still be saved. But I guess your way gives the user more control over what he can save and should the user decide that he or she isn't satisfied with what's been inputted, a simple restart of Arli will allow them to start over. Depends on what you want to emphasise on for Arli.